src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -350,4 +350,31 @@
     <ProjectCapability Include="RoslynComponent" Condition="'$(IsRoslynComponent)' == 'true'"/>
   </ItemGroup>
 
+  <!--
+    ========================
+    Read Compiler Written Files
+    ========================
+    
+    After compilation, read back the list of files the compiler wrote to disk
+    and add them to FileWrites so they are properly tracked by MSBuild
+    -->
+  <Target Name="ReadCompilerTouchedFilesBack" 
+          AfterTargets="CoreCompile"
+          Inputs="$(TouchedFilesOutputPath).write"
+          Outputs="$(TouchedFilesOutputPath).write"
```

Comment:
Unsure about this. It means the task doesn't run if the compiler didn't run, but I'm not sure if it'll give the correct results. Should we just always run this task?